app/src/app/framework/services/type.service.ts

```diff
@@ -62,7 +64,40 @@ export class TypeService<T = CmdbType> implements ApiService {
 
   public servicePrefix: string = 'types';
 
-  constructor(private api: ApiCallService) {
+  constructor(private api: ApiCallService, private userService: UserService) {
+  }
+
+  /**
+   * Returns acl read filter
+   *
+   * @private
+   */
+  private getAclReadFilter() {
+    const location = 'acl.groups.includes.' + this.userService.getCurrentUser().group_id;
+    return {
+      $or: [
+        { $or : [{acl: { $exists: false }}, {'acl.activated' : false}]},
+        { $and : [
+            {'acl.activated' : true},
+            {$and : [
+                { [location] : { $exists: true }},
+                { [location] : { $in : ['READ']}}
+              ]},
+          ]}]
+    };
+  }
+
+  /**
+   * Filters all types which don't include the READ right
+   *
+   * @param types @link{CmdbType[]}
+   * @param right
+   * @return @link{CmdbType[]}
+   */
+  public filterTypesByAcl(types: CmdbType[], right: AccessControlPermission): CmdbType[] {
```

Comment:
Ok, this is not a function that belongs in the type service. It is used for a specific selection of components. Either it is used on the component level itself or better yet, the data does not get into the frontend at all.